### ISSUE TITLE
org.jooq.Meta should try parsing default expressions

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#8469]
Create this table in PostgreSQL:

```sql
CREATE TABLE t(
  a bigint[] DEFAULT '{}'::bigint[]
);
```

Now access it via `org.jooq.Meta`:

```java
List<Schema> schemas = ctx.meta().getSchemas("public");
for (var schema : schemas) {
    System.out.println(schema.getTables());
}
```

#8468 shows an exception that will be ignored, but the fact is that jOOQ cannot currently recognise such expressions in the default expression value. We should use the parser to parse these expressions.

### Tasks

- [x] Add `Settings.parseMetaDefaultExpressions` to allow for opting out of this parsing
- [x] Implement parsing
- [x] Enable it for dialects that produce expressions in `DatabaseMetaData`, including:
  - [x] H2
  - [x] PostgreSQL

----
See also: 
- #8468 *"DataTypeException when org.jooq.Meta accesses PostgreSQL array column with default expression"*
- #8875 *"Field.getDataType().defaultValue() value is 'null' instead of actual value."*

**Comment by lukaseder** [comment]
We'll do this for dialects that report default expressions through `DatabaseMetaData`, instead of values (to be investigated). List in issue description will be amended

The parser call is protected by a try-catch block, reverting to storing the default expression as plain SQL and logging an info message. If that message gets too noisy, users can turn off the feature using a new flag: `Settings.parseMetaDefaultExpressions`.

**Comment by lukaseder** [comment]
Fix was included as part of https://github.com/jOOQ/jOOQ/commit/954bf584032f97c0e32b4a68be45cf3451df0a42.

Further dialects will still be fixed by 3.15.
